R/g.report.part5_dictionary.R

```diff
@@ -0,0 +1,252 @@
+        # Rest of column names...
+        if (is.null(what)) {
+          if ("sleeponset" %in% elements | "wakeup" %in% elements) {
+            if ("sleeponset" %in% elements) what = "Sleep onset time"
+            if ("wakeup" %in% elements) what = "Sleep onset time"
+            if ("ts" %in% elements) {
+              unit = "(hh:mm:ss)"
+            } else {
+              unit = "(hours from previous midnight)"
+            }
+          } else if ((substr(elements[1], 1, 1) == "L" | substr(elements[1], 1, 1) == "M")
+                     & elements[1] != "LUX") {
+            # LX and MX metrics
+            # what
+            if (grepl("TIME", elements[1])) {
+              what = "Starting time"
+              unit = "(timestamp)"
+              if ("num" %in% elements) {
+                unit = "(timestamp)"
+              }
+            } else if (grepl("VALUE", elements[1])) {
+              what = "Mean acceleration"
+              unit = "(mili-g units)"
+            } else if ("peakLUX" %in% elements) {
+              if ("mean" %in% elements) {
+                what = "Mean peak Lux"
+              } else if ("max" %in% elements) {
+                what = "Max peak Lux"
+              }
+            }
+            # class
+            X = as.numeric(gsub("\\D", "", elements))[1]
+            if (substr(elements[1], 1, 1) == "L") {
+              class = paste("during the", X, "consecutive hours with the lowest activity")
+            } else if (substr(elements[1], 1, 1) == "M") {
+              class = paste("during the", X, "consecutive hours with the highest activity")
+            }
+          } else if ("daytype" %in% elements) {
+            what = "WD = weekday; WE = weekend day"
+          } else {
+            what = ""
+          } 
+        } 
+      }
+      # build up definition
+      def = paste(what, class, window, unit)
+      # only for personsummary - aggregation method
+      if (grepl("personsummary", reports[ri])) {
+        agg = NULL
+        if (!grepl("Nvalid", cnames[coli])) {
+          if ("pla" %in% elements) agg = "- plain average"
+          if ("wei" %in% elements) agg = "- weighted average"
+          if ("WD" %in% elements) agg = "- weekdays average"
+          if ("WE" %in% elements) agg = "- weekend days average"
+        } else {
+          # adapt definition of valid days to part 5 reports
+          def = gsub("part 2, 4, and 5", "part 5", def)
+        }
+        def = paste(def, agg)
+      }
+      def = gsub("\\s+", " ", def) # this removes extra spaces
+      def = gsub("^ ", "", def) # this removes leading spaces
+      
+      # if window and class have not been identified, then return blank definition
+      if (is.null(what) & is.null(class)) def = NA
+      
+      # store definition
+      dictionary[coli, "Definition"] = def
+    }
+    # write csv reports with data dictionaries
+    directory = file.path(metadatadir, "results", "data dictionary/")
```

Comment:
Try to avoid spaces in filenames to avoid OS specific problems. Further, it is a variable dictionary not a data dictionary because everything is data.

So, maybe replace by folder name `variableDictionary`?